src/main/java/org/example/learningsystem/btp/saasprovisioningservice/service/SubscriptionServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package org.example.learningsystem.btp.saasprovisioningservice.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.example.learningsystem.btp.saasprovisioningservice.config.ApplicationProperties;
+import org.example.learningsystem.btp.saasprovisioningservice.dto.ServiceInfoDto;
+import org.example.learningsystem.btp.saasprovisioningservice.dto.SubscriptionRequestDto;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class SubscriptionServiceImpl implements SubscriptionService {
+
+    private static final String TENANT_SPECIFIC_URL_TEMPLATE = "https://%s-lms-approuter-dev.cfapps.us10-001.hana.ondemand.com";
```

Comment:
approuter name shouldn't be hardcoded and should be moved to environment variables and it can be placed in user-provided service (as it can be changed)
Additionally, space and domain can be changed and we should be prepared for this. It also can be done via environment variables and we can read it from VCAP_APPLICATION env (you can find in service bindings of your app) `${vcap.application.application_uris[0]}`
And then you can build tenant-specific URL:
```
String approuterUri = applicationUri.replace(applicationName, approuterName);
return "%s://%s-%s".formatted(HTTPS_PROTOCOL, tenantSubdomain, approuterUri);
```